### PR TITLE
Concept Entry:  Go Strings

### LINKED ISSUE #1226: [Concept Entry] Go: strings
### Reason/inspiration (optional)

We would like a new concept entry for [Go](https://www.codecademy.com/resources/docs/go) about strings.

The entry should go in a new file in docs/content/go/concepts/strings/strings.md

The entry should
- Explain the difference between Go strings and strings in other languages.
- Explain string literals and escapes using an example ```go code snippet.
- Explain the difference between literals using double quotes (") vs. literals using backticks (`)

### Entry Type (select all that apply)

New Concept Entry for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by Claudio-AhumadaL**
### Description

Closes #1226. It creates the concept entry of Go strings. The metadata and content are included in the strings.md file saved in ../docs/content/go/concepts/strings directory. The content included defining string literals, showcasing its differences with strings in other languages, and the description and use of escape characters.

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/go/concepts/strings/strings.md
Comment (SSwiniarski) [inline content/go/concepts/strings/strings.md:3]:
```suggestion
Description: 'A string is a read-only data type. Strings are chains of arbitrary bytes representing UTF-8 encoded characters.' 
```

File: content/go/concepts/strings/strings.md
Comment (SSwiniarski) [inline content/go/concepts/strings/strings.md:2]:
```suggestion
Title: 'Strings'
```

File: content/go/concepts/strings/strings.md
Comment (SSwiniarski) [inline content/go/concepts/strings/strings.md:65]:
```suggestion
Escape characters are useful to avoid ambiguities when the string you are creating has a character that may be problematic. An example of this would be defining `There is no "I" in TEAM` as an interpreted literal string. Since the string has double quotes in it, the compiler would throw an error message if you try to define it as it is. This can be avoided by using the double quote escape character ` \''`. Take a look at this example:
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@Claudio-AhumadaL I did a first review and left a few comments.

**Comment by Claudio-AhumadaL** [comment]
@SSwiniarski Hi, thank you for the review, I already updated the file with your suggestions.

File: content/go/concepts/strings/strings.md
Comment (SSwiniarski) [inline content/go/concepts/strings/strings.md:17]:
```suggestion
As in many other languages, strings in Go are sequences of characters. *Strings are immutable*. Once a Go string is created, its value cannot be changed. If you try to change it, the compiler will throw an error message. This is a main difference between strings in Go and strings in other languages.
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@Claudio-AhumadaL I had one more comment.

**Comment by Claudio-AhumadaL** [comment]
@SSwiniarski Hi! All done, I commited the most recent change too.

**Review by KTom101 — CHANGES_REQUESTED** [review]
@Claudio-Ahumadal I've added a second review to this entry.

**Comment by Claudio-AhumadaL** [comment]
Hi @KTom101, thank you for the review. I think I commited all the changes.

**Comment by Claudio-AhumadaL** [comment]
Hi! I noticed that there were 2 failing checks. I noticed that it has something to do with a deprecated command (save-state). How can I help fixing this issue? I don't know what can I do on my end to fix it, could you please point me to more information about it? I read [this entry](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/) so I think I understand the problem, but I can't figure out how to actually implement the solution and I don't want to just go ahead and try blindly because I don't want to cause new problems while doing it.

Thank you very much in advance and apologies for this issue.

**Comment by KTom101** [comment]
@Claudio-AhumadaL The warnings you see is something for us on our end. The warnings don't have anything to do with why the checks failed. 

It looks like the checks are failing due to formatting and lint. There are steps within the Contribution Guide [here](https://github.com/Codecademy/docs/blob/main/.github/CONTRIBUTING.md#i-submitted-a-pr-but-tests-are-failing-how-do-i-fix-them) to fix these tests. You will need to install `yarn` (as mentioned [here](https://github.com/Codecademy/docs/blob/main/.github/CONTRIBUTING.md#how-can-i-test-my-changes-before-submitting-a-pr))if you haven't done so already. 

Let me know if you're still having trouble after trying this. I can always do it on my end.

**Comment by Claudio-AhumadaL** [comment]
Hi @KTom101, how are you? I tried what you suggested (thank you very much!). yarn:format:verify doesn't throw an error anymore, but yarn lint:md still does. I tried fixing the errors displayed there (which were related to extra spaces in lines 55 to 65 (excluding line 60) and line 84). I saved the changes and made a commit, but the problem persists. I pushed the changes I made via GitHub Desktop so you can check it out, please, because I don't know how to fix it.

I hope I didn't made any mistakes, this is my first contribution and I started coding very recently so I'm still on that phase where you are nervous at every step.

Let me know if there is any more I can do.

Thank you again for your help.

**Review by KTom101 — APPROVED** [review]
@Claudio-AhumadaL I've run the formatting and linting on the file and the checks are now passing. For future reference, I used `npm run format:all` to do this. It looks like in this case the issue was mostly coming from the table due to hard tabs being used instead of spacing. 

I also removed the italics from the file since we try to stay away from using them but I kept the introduced terms "interpreted string literal" and "raw string literal" in bold

**Comment by KTom101** [comment]
@SSwiniarski Is there anything you think should be added to this before merging?

**Comment by yangc95** [comment]
I made a small change [here](https://github.com/Codecademy/docs/pull/1535/commits/4ad486011f6f1d6cf133cd923d1b7b123e480e94) @KTom101 @SSwiniarski

Post-merge, I think if may be good to create issues for any missing data types as well as an edit to the data types entry to link out to this entry. Looks like lots of terms can be added for each of these types so that's a huge plus 😄

**Comment by SSwiniarski** [comment]
> @SSwiniarski Is there anything you think should be added to this before merging?

@KTom101 looks good to me.

**Comment by KTom101** [comment]
@Claudio-AhumadaL Thanks for contributing! The link is here: https://www.codecademy.com/resources/docs/go/strings 🙌 Please feel free to look at the [Issues Tab](https://github.com/Codecademy/docs/issues) for other contributing opportunities or let us know if you want to add something new to Codecademy Docs!

Screen Shot 2023-02-08 at 6 20 37 PM

**Comment by Claudio-AhumadaL** [comment]
Hi @KTom101. Great! This was very fun to work in, thank you for your help (and thanks to @SSwiniarski and @yangc95 too!). 
Best regards.

**Pull request merged**
